pelican/plugins/sitemap/sitemap.py

```diff
@@ -55,235 +58,104 @@ def format_date(date):
     return date.strftime("%Y-%m-%dT%H:%M:%S") + tz
 
 
-class SitemapGenerator(object):
-    def __init__(self, context, settings, path, theme, output_path, *null):
-
-        self.output_path = output_path
-        self.context = context
-        self.now = datetime.now()
-        self.siteurl = settings.get("SITEURL")
+def _get_format(settings):
+    fmt = settings.get("SITEMAP", {}).get("format", "xml")
+    assert fmt in ("txt", "xml")
```

Comment:
`assert` is not recommended for control flow, e.g. you can skip `assert`s with `python -O`. Just check and raise an actual exception. (`ValueError`?)